src/JuliusSweetland.OptiKey.Core/UI/ViewModels/Management/GesturesViewModel.cs

```diff
@@ -115,24 +119,35 @@ private void Load()
             EyeGesturesEnabled = Settings.Default.EyeGesturesEnabled;
             EyeGestureFile = Settings.Default.EyeGestureFile;
             EyeGestureString = Settings.Default.EyeGestureString;
-            xmlEyeGestures = XmlEyeGestures.ReadFromString(EyeGestureString);
+            xmlEyeGestures = XmlEyeGestures.ReadFromString(EyeGestureString) ?? new XmlEyeGestures();
             EyeGesture = GestureList != null && GestureList.Any() ? GestureList[0] : null;
         }
 
         private void OpenFile()
         {
             var fileDialog = new System.Windows.Forms.OpenFileDialog() { FileName = Path.GetFileName(EyeGestureFile) };
             var result = fileDialog.ShowDialog();
+            string tempFilename;
             switch (result)
             {
                 case System.Windows.Forms.DialogResult.OK:
-                    EyeGestureFile = fileDialog.FileName;
+                    tempFilename = fileDialog.FileName; // we will commit it if loading is okay                    
                     break;
                 case System.Windows.Forms.DialogResult.Cancel:
                 default:
-                    break;
+                    return;
+            }
+            try
+            {
+                xmlEyeGestures = XmlEyeGestures.ReadFromFile(tempFilename);                
+            }
+            catch (Exception e)
+            {
+                Log.Error($"Error reading from gesture file: {EyeGestureFile} :");
```

Comment:
Should this not be logging the tempFilename as the EyeGestureFile has not yet been updated?

Reply:
yes good spot